dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryDirectory.java

```diff
@@ -354,8 +356,11 @@ private URL mergeUrl(URL providerUrl) {
 
         providerUrl = providerUrl.addParameter(Constants.CHECK_KEY, String.valueOf(false)); // Do not check whether the connection is successful or not, always create Invoker!
 
+        // fix issue#9922
+        Map<String, String> providerSideParameters = new HashMap<>(providerUrl.getParameters());
+        providerSideParameters.remove(CommonConstants.TAG_KEY);
         // The combination of directoryUrl and override is at the end of notify, which can't be handled here
-        this.overrideDirectoryUrl = this.overrideDirectoryUrl.addParametersIfAbsent(providerUrl.getParameters()); // Merge the provider side parameters
+        this.overrideDirectoryUrl = this.overrideDirectoryUrl.addParametersIfAbsent(providerSideParameters); // Merge the provider side parameters

```

Comment:
`DefaultProviderURLMergeProcessor` has removed `dubbo.tag` , It's not the best way to fix it
.

Reply:
`DefaultProviderURLMergeProcessor` only remove consumer side `dubbo.tag`, provder side `dubbo.tag` still exist